br/pkg/task/restore.go

```diff
@@ -924,10 +894,16 @@ func runRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 		}
 	}
 
+	// preallocate the table id, because any ddl job or database creation(include checkpoint) also allocates the global ID
```

Comment:
I just wondering how to catch this issue?

Reply:
Actually, there are many orders between the operations in `runRestore`.
For example, we have the order `Get TableInfos From BackupMeta` -> `Pre-Allocate Table IDs`.
After the PR, we have the new order `Get TableInfos From BackupMeta` -> `Pre-Allocate Table IDs` -> `Save Checkpoint Meta Table` (create database[__Temporary_...._Snapshot_Checkpoint]/table[cpt_meta]). That's because any DDL jobs before `preallocation` will occupy the ID of upstream tables.

To catch this issue, maybe we need to make the preallocate more early.